backend/src/indexer/block_preprocessor.rs

```diff
@@ -291,36 +299,114 @@ impl concordium_rust_sdk::indexer::Indexer for BlockPreProcessor {
 /// Compute the total stake capital by summing all the stake of the bakers.
 /// This is only needed for older blocks, which does not provide this
 /// information as part of the tokenomics info query.
-pub async fn compute_total_stake_capital(
+pub async fn compute_validator_staking_information(
     client: &mut v2::Client,
     block_height: v2::BlockIdentifier,
-) -> v2::QueryResult<Amount> {
-    let mut total_staked_capital = Amount::zero();
+) -> v2::QueryResult<(Amount, Vec<(i64, ValidatorStakingInformation)>)> {
+    let mut all_pool_total_capital = Amount::zero();
     let mut bakers = client.get_baker_list(block_height).await?.response;
+    let mut validator_staking_info_vector: Vec<(i64, ValidatorStakingInformation)> = Vec::new();
+
     while let Some(baker_id) = bakers.try_next().await? {
-        let account_info = client
-            .get_account_info(&v2::AccountIdentifier::Index(baker_id.id), block_height)
-            .await?
-            .response;
-        total_staked_capital += account_info
-            .account_stake
-            .context("Expected baker to have account stake information")
-            .map_err(v2::RPCError::ParseError)?
-            .staked_amount();
+        // Try to get the pool info for a given Baker
+        if let Ok(baker_pool_info) = client.get_pool_info(block_height, baker_id).await {
```

Comment:
As this function `get_pool_info` only returns data starting from protocol version 4. I think we should add a comment here why we need to split into two cases and mention that the split happen at protocol version 4.

   /// Get information about a given pool at the end of a given block.
    /// If the block does not exist or is prior to protocol version 4 then
    /// [`QueryError::NotFound`] is returned.